src/software/ai/navigator/navigator.cpp

```diff
@@ -317,19 +326,64 @@ double Navigator::getCloseToEnemyObstacleFactor(const Point &p)
         }
     }
 
-    // linear mapping of 0 to 1 onto 0 to 2 and return 1 when closest_dist is greater than
-    // 2
-    if (closest_dist > 2)
+    // clamp ratio between 0 and 1
+    return std::clamp(closest_dist / ROBOT_MAX_SPEED_METERS_PER_SECOND, 0.0, 1.0);
+}
+
+void Navigator::processPathIntoCurrentPrimitive(std::optional<Path> path,
```

Comment:
Perhaps we could rename this to `getPrimitiveFromPath` instead and have it return a primitive? I think that would be much more clear. Personally, I had to read this a few times to figure out what was going on.

Reply:
done